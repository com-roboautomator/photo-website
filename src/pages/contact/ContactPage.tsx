import {createStyles, withStyles, WithStyles} from '@material-ui/styles'
import React from 'react'
import PageTitle from '../components/text/PageTitle'
import InputField from './InputField'
import SubmitButton from './SubmitButton'
import {sendEmail} from '../../email/Mailer'
import {RouterProps} from 'react-router-dom'

interface ContactProps extends WithStyles<typeof styles> {
    // for testing purposes. However may become useful for prefilled form
    presetName?: string
    presetAddress?: string
    presetMessage?: string
}

class Contact extends React.Component<ContactProps & RouterProps, any> {
    constructor(props: ContactProps & RouterProps) {
        super(props)
        this.state = {
            name: this.props.presetName,
            address: this.props.presetAddress,
            message: this.props.presetMessage,

            nameError: '',
            addressError: '',
            messageError: '',
        }
    }

    render() {
        const classes = this.props.classes
        return (
            <main data-testid="Contact-Page">
                <PageTitle
                    title="CONTACT"
                    src="https://doneux-website.s3.eu-west-2.amazonaws.com/fd1f733e-2cdf-46ec-b1eb-3e5e55a98a93.jpg"
                />
                <div className={classes.container}>
                    <div className={classes.wrapper}>
                        <p className={classes.header}>
                            <b>Contact Me</b>
                            <br />
                            <br />
                            Please fill in this form and I will do my best to
                            respond as soon as I can
                        </p>
                        <InputField
                            errorText={this.state.nameError}
                            onChange={this.handleNameUpdate.bind(this)}
                            label="Full Name"
                            value={this.state.name}
                        />
                        <InputField
                            errorText={this.state.addressError}
                            onChange={this.handleAddressUpdate.bind(this)}
                            label="Email Address"
                            value={this.state.address}
                        />
                        <InputField
                            errorText={this.state.messageError}
                            onChange={this.handleMessageUpdate.bind(this)}
                            label="Message"
                            value={this.state.message}
                            rows={8}
                        />
                        <SubmitButton onSubmit={this.handleSubmit.bind(this)} />
                    </div>
                </div>
            </main>
        )
    }

    nextPath(path: string | undefined) {
        if (path === undefined) return
        this.props.history.push(path)
    }

    handleNameUpdate(inputName: string) {
        this.setState({
            name: inputName ? inputName : this.state.name,
        })
    }

    handleAddressUpdate(inputAddress: string) {
        this.setState({
            address: inputAddress ? inputAddress : this.state.address,
        })
    }

    handleMessageUpdate(inputMessage: string) {
        this.setState({
            message: inputMessage ? inputMessage : this.state.inputMessage,
        })
    }

    handleSubmit() {
        const nameErrorUpdate =
            this.state.name === '' || this.state.name === undefined
                ? 'required'
                : ''
        const addressErrorUpdate =
            this.state.address === '' || this.state.address === undefined
                ? 'required'
                : ''
        const messageErrorUpdate =
            this.state.message === '' || this.state.message === undefined
                ? 'required'
                : ''

        this.setState({
            nameError: nameErrorUpdate,
            addressError: addressErrorUpdate,
            messageError: messageErrorUpdate,
        })

        if (
            nameErrorUpdate === '' &&
            addressErrorUpdate === '' &&
            messageErrorUpdate === ''
        ) {
            //email submission has been successful - update page to contact, reset values and confirmation then send email
            this.nextPath('/contact/confirmation')
            // resetting values is important for tests
            this.setState({
                name: '',
                address: '',
                message: '',
            })
            sendEmail(this.state.name, this.state.address, this.state.message)
        }
    }
}

const styles = () =>
    createStyles({
        container: {
            display: 'flex',
            justifyContent: 'center',
        },
        wrapper: {
            width: '40%',
            display: 'flex',
            flexDirection: 'column',
            paddingTop: '20px',
        },
        header: {
            fontFamily: 'Open-Sans-Regular',
            fontSize: '18px',
            width: '80%',
            alignSelf: 'center',
        },
    })

export default withStyles(styles)(Contact)
