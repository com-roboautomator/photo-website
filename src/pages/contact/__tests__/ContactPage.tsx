import '@testing-library/jest-dom/extend-expect'
import {render} from '@testing-library/react'
import {shallow} from 'enzyme'
import {BrowserRouter, Route, Router} from 'react-router-dom'
import renderer from 'react-test-renderer'
import ContactPage from '../ContactPage'

describe('Contact Page', () => {
    it('should render correctly when there are no items', () => {
        const ImageSliderRender = renderer
            .create(
                <BrowserRouter>
                    <Route
                        render={(props: any) => <ContactPage {...props} />}
                    />
                </BrowserRouter>
            )
            .toJSON()

        expect(ImageSliderRender).toMatchSnapshot()
    })

    it('should render correctly', () => {
        const {getByTestId} = render(
            <BrowserRouter>
                <Route render={(props: any) => <ContactPage {...props} />} />
            </BrowserRouter>
        )

        expect(
            getByTestId('InputField-TextField-Full Name')
        ).toBeInTheDocument()
        expect(
            getByTestId('InputField-TextField-Email Address')
        ).toBeInTheDocument()
        expect(getByTestId('InputField-TextField-Message')).toBeInTheDocument()

        expect(getByTestId('Submit-Button')).toBeInTheDocument()
    })
})
