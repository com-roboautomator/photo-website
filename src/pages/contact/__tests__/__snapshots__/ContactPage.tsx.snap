// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Page should render correctly when there are no items 1`] = `
<main
  data-testid="Contact-Page"
>
  <div
    className="PageTitle-container-4"
    data-testid="PageTitle-Container"
  >
    <img
      alt=""
      className="PageTitle-image_loading-6"
      data-testid="PageTitle-Image"
      onLoad={[Function]}
      src="https://picsum.photos/id/1058/1500/200"
      style={
        Object {
          "height": "20vh",
        }
      }
    />
    <div
      className="PageTitle-title_loading-7"
      data-testid="PageTitle-Text"
    >
      CONTACT
    </div>
    <div
      className="PageTitle-signature-9"
    >
      <img
        alt=""
        data-testid="PageTitle-Signature"
        src=""
        width="25%"
      />
    </div>
  </div>
  <div
    className="Contact-container-1"
  >
    <div
      className="Contact-wrapper-2"
    >
      <p
        className="Contact-header-3"
      >
        <b>
          Contact Me
        </b>
        <br />
        <br />
        Please fill in this form and I will do my best to respond as soon as I can
      </p>
      <div
        className="InputField-container-10"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            data-testid="InputField-TextField-Full Name"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Full Name
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-13"
                >
                  <span>
                    Full Name
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="InputField-container-10"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            data-testid="InputField-TextField-Email Address"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Email Address
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="text"
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-13"
                >
                  <span>
                    Email Address
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="InputField-container-10"
      >
        <div
          className="MuiFormControl-root MuiFormControl-fullWidth"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
            data-testid="InputField-TextField-Message"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
              data-shrink={false}
            >
              Message
            </label>
            <div
              className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
              onClick={[Function]}
            >
              <textarea
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
                disabled={false}
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                rows={8}
              />
              <fieldset
                aria-hidden={true}
                className="PrivateNotchedOutline-root-11 MuiOutlinedInput-notchedOutline"
              >
                <legend
                  className="PrivateNotchedOutline-legendLabelled-13"
                >
                  <span>
                    Message
                  </span>
                </legend>
              </fieldset>
            </div>
            
          </div>
        </div>
      </div>
      <div
        className="SubmitButton-container-15"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained SubmitButton-button-16 MuiButton-containedPrimary"
          data-testid="Submit-Button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiButton-label"
          >
            Submit
          </span>
        </button>
      </div>
    </div>
  </div>
</main>
`;
